src/services/game-logic/GameEngine.ts

```diff
@@ -12,16 +12,141 @@ class GameEngine {
 
     getLegalMoves = (square: Square): Square[] => {
         const piece = this.board.getPiece(square);
-        let pieceMoves = piece?.getPossibleMoves(this.board) ?? [];
-        return pieceMoves.filter(this.isOnBoard).filter((square) => !this.isOccupiedBySameColor(square, piece));
+        const pieceMoves = piece!.getPossibleMoves(this.board);
+
+        const firstCollission = this.getFirstCollission(square);
+        const pieceMoves2 = pieceMoves
```

Comment:
weź jakoś zmień te nazwę

Reply:
i ogólnie nie rozumiem idei jaka stoi za stworzeniem zmiennych pieceMoves i pieceMoves2... chyba znaczące i tak są tylko pieceMoves2... ale może się mylę?

Reply:
poprawione już na legalMoves